eval.c

```diff
@@ -1079,11 +1079,21 @@ rb_mod_prepend(int argc, VALUE *argv, VALUE module)
 {
     int i;
     ID id_prepend_features, id_prepended;
+    VALUE new_mod;
 
     CONST_ID(id_prepend_features, "prepend_features");
     CONST_ID(id_prepended, "prepended");
 
-    rb_check_arity(argc, 1, UNLIMITED_ARGUMENTS);
+    if(rb_block_given_p()) {
```

Comment:
I added another test showing this behavior in https://github.com/ruby/ruby/pull/1639/commits/3331f39b58816b1258c820ff9820550717697fe0